sdcm/sct_events/base.py

```diff
@@ -249,25 +256,77 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.end_event()
         return self
 
+    @property
+    def msgfmt(self):
+        fmt = super().msgfmt
+        LOGGER.info(f"msgfmt: {self.duration}")
+        if self.duration is not None:
+            fmt += " duration={0.duration_formatted}"
+        return fmt
+
     @property
     def errors_formatted(self):
         return "\n".join(self.errors) if self.errors is not None else ""
 
+    @property
+    def duration(self):
+        # Event filter does not create object of the class (not initialize it), so "_duration" attribute does not
+        # exist there and filter fails
+        if not hasattr(self, "_duration"):
+            LOGGER.info(f"duration: _duration attribute does not exist")
+            LOGGER.info(f"duration attributes: {self.__dict__}")
+            return None
+        LOGGER.info(f"duration: {self.begin_timestamp} {self.end_timestamp} {self._duration}")
+        if self._duration is None:
+            if self.begin_timestamp is not None and self.end_timestamp is not None:
+                self._duration = int(self.end_timestamp - self.begin_timestamp)
+        LOGGER.info(f"duration._duration: {self._duration}")
+        return self._duration
+
+    @duration.setter
+    def duration(self, duration: int):
+        self._duration = duration
+
+    @property
+    def duration_formatted(self):
```

Comment:
Would be really good to have unit tests for this "duration_formatted" method. It is something that is easy to break silently.

Reply:
Actually it's tested in the every test, when `str(event)` is performed.
I created in the every test different situations and test different cases

Reply:
Exactly, that is why it must be done in one place. To reduce the obscurity of it's coverage and forget about the need to use different values in it's consumers.

Reply:
added test to `unit_tests/test_events.py`